src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +52,74 @@ public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final List<String>
     }
   }
 
-
+  /**
+   * Get the historical traded prices for a given range or date.
+   *
+   * @param symbols the ticker for the stock
+   * @param ranges the time series for the historical traded price (max,5y,2y,1y,ytd,6m,3m,1m,5d)
+   * @param date the specific date (YYYYMMDD)
+   * @return A list of historical traded price objects for each Symbol that is passed in
+   */
+  public List<List<IexHistoricalPrice>> getHistoricalPriceForSymbol(
```

Comment:
This method is very long and contains multiple levels of braces. Per clean code best practices, try to keep methods around 5 lines or less, and extract multiple nested braces into individual named methods.